File: problemsets/Codeforces/C++/A127.cpp
/**
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 */
#include <bits/stdc++.h>
using namespace std;
int main() {
  int n, k; cin >> n >> k;
  auto a = vector<pair<int,int>>(n);
  for (auto &u: a) cin >> u.first >> u.second;
  double t = 0;
  auto dist = [](pair<int,int> u,pair<int,int> v) {
    int dx = u.first - v.first;
    int dy = u.second - v.second;
    return sqrt(dx*dx + dy*dy);
  };
  for (int i = 1; i < n; i++) t += dist(a[i],a[i-1]) / 50.;
  printf("%.8lf\n", t*k);
}

File: problemsets/Codeforces/Python/A127.py
"""
 *
 * Author:  Juarez Paulino(coderemite)
 * Email: juarez.paulino@gmail.com
 *
 """
I=lambda:map(int,input().split())
n,k=I()
a=[[*map(int,input().split())]for _ in[0]*n]
r=0
for u,v in zip(a,a[1:]):
  r+=((u[0]-v[0])**2+(u[1]-v[1])**2)**0.5
print(r*k/50)